app/src/main/java/com/bignerdranch/android/geoquiz/MainActivity.kt

```diff
@@ -3,64 +3,144 @@ package com.bignerdranch.android.geoquiz
+private const val KEY_SCORE = "score"
+private const val KEY_TOTALANSWERS = "total_answers"
+
 class MainActivity : AppCompatActivity() {
+
     private lateinit var binding: ActivityMainBinding
-    private var currentIndex = 0
+    private val quizViewModel: QuizViewModel by lazy {
+        ViewModelProviders.of(this).get(QuizViewModel::class.java)
+    }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setupViewBinding()
+        getSavedInstanceState(savedInstanceState)
         setupClickListener()
+        checkQuestion()
         updateQuestion()
     }
 
+    override fun onSaveInstanceState(savedInstanceState: Bundle) {
+        super.onSaveInstanceState(savedInstanceState)
+        savedInstanceState.putInt(KEY_INDEX, quizViewModel.currentIndex)
+        savedInstanceState.putInt(KEY_SCORE, quizViewModel.score)
+        savedInstanceState.putInt(KEY_TOTALANSWERS, quizViewModel.answersCounter)
+    }
+
+    private fun setupViewBinding() {
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        val view = binding.root
+        setContentView(view)
+    }
+
+    private fun getSavedInstanceState(savedInstanceState: Bundle?) {
+        val currentIndex = savedInstanceState?.getInt(KEY_INDEX, 0) ?: 0
+        quizViewModel.currentIndex = currentIndex
+        val score = savedInstanceState?.getInt(KEY_SCORE, 0) ?: 0
+        quizViewModel.score = score
+        val totalAnswers = savedInstanceState?.getInt(KEY_TOTALANSWERS, 0) ?: 0
+        quizViewModel.answersCounter = totalAnswers
+    }
+
     private fun setupClickListener() {
         binding.trueButton.setOnClickListener { view: View ->
+            setAnswer()
             checkAnswer(true)
+            checkResult()
         }
         binding.falseButton.setOnClickListener { view: View ->
+            setAnswer()
             checkAnswer(false)
+            checkResult()
         }
         binding.nextButton.setOnClickListener {
-            currentIndex = (currentIndex + 1) % QuestionsBank.getBankSize()
+            moveToNext()
+            checkQuestion()
             updateQuestion()
         }
         binding.prevButton.setOnClickListener {
-            if (currentIndex - 1 < 0) {
-                currentIndex = QuestionsBank.getBankSize() - 1
-            } else {
-                currentIndex = (currentIndex - 1) % QuestionsBank.getBankSize()
-            }
+            moveToPrev()
+            checkQuestion()
             updateQuestion()
         }
         binding.questionTextView.setOnClickListener { view: View ->
-            currentIndex = (currentIndex + 1) % QuestionsBank.getBankSize()
+            moveToNext()
+            checkQuestion()
             updateQuestion()
         }
     }
 
-    private fun setupViewBinding() {
-        binding = ActivityMainBinding.inflate(layoutInflater)
-        val view = binding.root
-        setContentView(view)
+    private fun checkQuestion() {
```

Comment:
the name should say what it do, setupAnswerButtonState or something like that

Reply:
All fixed!